### ISSUE TITLE
Allow specifying the alignment of extern structs

### ISSUE THREAD

**Issue by wooster0** [ziglang/zig#13009]
This is a follow-up to #10113: https://github.com/ziglang/zig/issues/10113#issuecomment-1241789307
We need an addition to the language that allows us to specify the alignment of `extern struct`s.
The UEFI libs are particularly broken because since #10113 any packed struct with an array or any extern struct in it isn't a valid packed type.
In order to make `std.os.uefi` work with self-hosted, #12899 was an effort to make it easier to add `align(1)` to every field of a struct. The problem is that it might not work that well considering that [structs with decls can't be reified](https://github.com/ziglang/zig/pull/12899#issuecomment-1261425533) because of #6709.
But instead of allowing #6709 (which might be a bad idea) and then using `CPacked`, it's probably best to allow specifying the alignment of `extern struct`s at the language level.

A good syntax for this would be `extern struct align(alignment) {}`.

See also: #12745 #12897 #12993

CC @fifty-six

**Comment by topolarity** [comment]
To clarify, this would specify the default alignment of every field within the struct?
(i.e., the effective alignment of the extern struct might differ after explicit field alignments are applied?)

Somewhat related, esp. regarding syntax: https://github.com/ziglang/zig/issues/8643

**Comment by r00ster91** [comment]
Yes, that's basically what it does. It's like applying that same alignment manually to every field in the struct.

**Comment by daurnimator** [comment]
> To clarify, this would specify the default alignment of every field within the struct?

I wouldn't expect that from the syntax: i would expect it to be the minimum alignment for the entire struct (i.e. the first field of the struct)

**Comment by sDIMMaX** [comment]
Not sure, but probality can use union and set align to all struct as some hack:
```
pub const Char = extern union {
    const CHAR_SIZE = 4;

    raw: u64 align(8),

    char: extern struct {
        data: extern union {
            raw: u32 align(1),
            u8x4null: [CHAR_SIZE:0]u8,
        },
        _flags: u8,
        lang: u16,
    },
}
```
and `std.testing.expectEqual(8, @alignOf(Char)) catch unreachable;` technicaly say `raw: u64 align(8)` works.
for me this works. `std.testing.expectEqual(8, @sizeOf(Char)) catch unreachable;`
